lib/rucio/daemons/bb8/common.py

```diff
@@ -221,100 +264,239 @@ def list_rebalance_rule_candidates(rse_id, mode=None, session=None):
+    only_move_closed_did = config_get_bool(
+        section="bb8",
+        option="only_move_closed_did",
+        raise_exception=False,
+        default=None,
+        expiration_time=3600,
+    )
     if only_move_closed_did:
         did_clause.append(models.DataIdentifier.is_open == False)  # NOQA
 
     # Now build the query
     external_dsl = aliased(models.DatasetLock)
-    count_locks = select([func.count()]).where(and_(external_dsl.scope == models.DatasetLock.scope,
-                                                    external_dsl.name == models.DatasetLock.name,
-                                                    external_dsl.rse_id == models.DatasetLock.rse_id)).as_scalar()
-    query = session.query(models.DatasetLock.scope,
-                          models.DatasetLock.name,
-                          models.ReplicationRule.id,
-                          models.ReplicationRule.rse_expression,
-                          models.ReplicationRule.subscription_id,
-                          models.DataIdentifier.bytes,
-                          models.DataIdentifier.length,
-                          case([(or_(models.DatasetLock.length < 1, models.DatasetLock.length.is_(None)), 0)],
-                               else_=cast(models.DatasetLock.bytes / models.DatasetLock.length, BigInteger))).\
-        join(models.ReplicationRule, models.ReplicationRule.id == models.DatasetLock.rule_id).\
-        join(models.DataIdentifier, and_(models.DatasetLock.scope == models.DataIdentifier.scope, models.DatasetLock.name == models.DataIdentifier.name)).\
-        filter(models.DatasetLock.rse_id == rse_id).\
-        filter(and_(*rule_clause)).\
-        filter(and_(*did_clause)).\
-        filter(case([(or_(models.DatasetLock.length < 1, models.DatasetLock.length.is_(None)), 0)],
-                    else_=cast(models.DatasetLock.bytes / models.DatasetLock.length, BigInteger)) > 1000000000).\
-        filter(count_locks == 1)
-    summary = query.order_by(case([(or_(models.DatasetLock.length < 1, models.DatasetLock.length.is_(None)), 0)],
-                                  else_=cast(models.DatasetLock.bytes / models.DatasetLock.length, BigInteger)),
-                             models.DatasetLock.accessed_at).all()
+    count_locks = (
+        select([func.count()])
+        .where(
+            and_(
+                external_dsl.scope == models.DatasetLock.scope,
+                external_dsl.name == models.DatasetLock.name,
+                external_dsl.rse_id == models.DatasetLock.rse_id,
+            )
+        )
+        .as_scalar()
+    )
+    query = (
+        session.query(
+            models.DatasetLock.scope,
+            models.DatasetLock.name,
+            models.ReplicationRule.id,
+            models.ReplicationRule.rse_expression,
+            models.ReplicationRule.subscription_id,
+            models.DataIdentifier.bytes,
+            models.DataIdentifier.length,
+            case(
```

Comment:
this new formatting makes me want to tear my eyes off

Reply:
I agree at least in this case.... usually I don't mind so much but this introduces way too much verticality. Much easier to understand what's going on if you don't have to scroll.

Reply:
That's much better on my 4:3 display :trollface:

Reply:
+1 I find that one also extremely unreadable 😢